code/gui/serialbox.h

```diff
@@ -5,6 +5,7 @@
 
 #include <QDialog>
 #include <QSlider>
+#include <QLineEdit>
```

Comment:
if you don't need the type definition in the header use a forward declaration
```c++
class QSlider;
class QLineEdit;
```

and then just include the appropriate header in the source file only